packages/temporal/mpe/workflows/mpe.go

```diff
@@ -161,6 +168,84 @@ func MpeRoomWorkflow(ctx workflow.Context, params shared_mpe.MpeRoomParameters)
 					},
 				},
 			},
+
+			MpeRoomReady: &brainy.StateNode{
+				On: brainy.Events{
+					MpeRoomAddTracksEventType: brainy.Transition{
+						Actions: brainy.Actions{
+							brainy.ActionFn(
+								func(c brainy.Context, e brainy.Event) error {
+									event := e.(MpeRoomAddTracksEvent)
+
+									acceptedTracksIDsToAdd := make([]string, 0, len(event.TracksIDs))
+
+									for _, trackToAdd := range event.TracksIDs {
+										isDuplicate := internalState.Tracks.Has(trackToAdd)
+										if isDuplicate {
+											continue
+										}
+
+										acceptedTracksIDsToAdd = append(acceptedTracksIDsToAdd, trackToAdd)
+									}
+
+									noTracksHaveBeenAccepted := len(acceptedTracksIDsToAdd) == 0
+									if noTracksHaveBeenAccepted {
+										sendRejectAddingTracksActivity(ctx, activities_mpe.RejectAddingTracksActivityArgs{
+											DeviceID: event.DeviceID,
+										})
+
+										return nil
+									}
+
+									fetchingFuture := sendFetchTracksInformationActivityAndForwardInitiator(
+										ctx,
+										acceptedTracksIDsToAdd,
+										event.UserID,
+										event.DeviceID,
+									)
+									fetchedAddedTracksInformationFutures = append(fetchedAddedTracksInformationFutures, fetchingFuture)
+
+									return nil
+								},
+							),
+						},
+					},
+
+					MpeRoomAddedTracksInformationFetchedEventType: brainy.Transition{
+						Actions: brainy.Actions{
+							brainy.ActionFn(
+								func(c brainy.Context, e brainy.Event) error {
+									event := e.(MpeRoomAddedTracksInformationFetchedEvent)
+
+									// If one track to add is already in the playlist, abort the whole operation.
+									// In practice, a single track can be added at a time.
+									// Therefore, aborting the whole operation makes sense.
```

Comment:
I totally understand this point of view, this is completly ok.
I don't know if it's still the best behaviour. Even if we're not expecting any user to be able to send a tracks array, Temporal is agnostic about that. Maybe should we throw an error only if none of the fetched tracks could be added ?

Reply:
It would be better, yeah!